spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/mapping/event/AfterDeleteEvent.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.datastore.core.mapping.event;
+
+import com.google.cloud.datastore.Key;
+
+/**
+ * An event published just after Spring Data Cloud Datastore performs a delete operation.
+ *
+ * @author Chengyuan Zhao
+ */
+public class AfterDeleteEvent extends DeleteEvent {
+
+	/**
+	 * Constructor.
+	 *
+	 * @param keysToDelete The keys that are deleted in this operation (never {@code null}).
+	 * @param targetEntityClass The target entity type deleted. This may be {@code null}
+	 *     depending on the specific delete operation.
+	 * @param targetIds The target entity ID values deleted. This may be {@code null}
+	 *     depending on the specific delete operation.
+	 * @param targetEntities The target entity objects deleted. This may be {@code null}
+	 *     depending on the specific
+	 */
+	public AfterDeleteEvent(Key[] keysToDelete, Class targetEntityClass, Iterable targetIds, Iterable targetEntities) {
```

Comment:
I'd avoid any nullable fields and instead have separate event classes to represent different delete events.
Otherwise, let's just keep the delete event simple and restrict the data in it to keys and entityClass.

Reply:
There are several issues with breaking it up. The reason is there are many delete operations where only 1 of the 4 fields are null. Several of the fields appear populated in several different delete operations.

In other words the delete operations overlap each other in which fields are populated and there wouldn't be a good way to give meaningful names to the delete events.  The fields and delete operations they correspond to are NOT disjoint.

Reply:
You shouldn't have more events than delete methods and using the delete method name to inform the name of the event seems to make sense.

Reply:
@meltsufin the other modules all have single pair of "BeforeDeleteEvent" and "AfterDeleteEvent". 
Also, because of the before/after pair, we would have TEN CLASSES for the 5 delete methods.

Reply:
Agreed the number of classes doubling is inconvenient.
Let's at least start by moving the nullable fields out of the constructor. The constructor should only take required fields. Also, it might make sense to provide `getOptional....` methods for nullable fields.

Reply:
Sounds good. Let me go ahead and do that.

Reply:
Done, PTAL @meltsufin